rdkit/Chem/Draw/UnitTestDraw.py

```diff
@@ -8,185 +7,154 @@
     img = Draw.MolToQPixmap(self.mol, size=(300, 300))
     self.assertTrue(img)
     self.assertEqual(img.size().height(), 300)
     self.assertEqual(img.size().width(), 300)
 
+  @unittest.skipIf(cairoCanvas is None, 'Skipping cairo test')
   def testCairoImageDash(self):
-    try:
-      from rdkit.Chem.Draw.cairoCanvas import Canvas
-    except ImportError:
-      return
     os.environ['RDKIT_CANVAS'] = 'cairo'
+    self._testMolToImage(kekulize=False)
 
-    img = Draw.MolToImage(self.mol, size=(300, 300), kekulize=False)
-    self.assertTrue(img)
-    self.assertEqual(img.size[0], 300)
-    self.assertEqual(img.size[1], 300)
-
+  @unittest.skipIf(aggCanvas is None, 'Skipping agg test')
   def testAggImageDash(self):
-    try:
-      from rdkit.Chem.Draw.aggCanvas import Canvas
-    except ImportError:
-      return
     os.environ['RDKIT_CANVAS'] = 'agg'
-    img = Draw.MolToImage(self.mol, size=(300, 300), kekulize=False)
-    self.assertTrue(img)
-    self.assertEqual(img.size[0], 300)
-    self.assertEqual(img.size[1], 300)
+    self._testMolToImage(kekulize=False)
 
+  @unittest.skipIf(spingCanvas is None, 'Skipping sping test')
   def testSpingImageDash(self):
-    try:
-      from rdkit.Chem.Draw.spingCanvas import Canvas
-    except ImportError:
-      return
     os.environ['RDKIT_CANVAS'] = 'sping'
-    img = Draw.MolToImage(self.mol, size=(300, 300), kekulize=False)
-    self.assertTrue(img)
-    self.assertEqual(img.size[0], 300)
-    self.assertEqual(img.size[1], 300)
+    self._testMolToImage(kekulize=False, showImage=False)
 
+  @unittest.skipIf(spingCanvas is None, 'Skipping sping test')
   def testGithubIssue54(self):
-    try:
-      from rdkit.Chem.Draw.spingCanvas import Canvas
-    except ImportError:
-      return
+    # Assert that radicals depict with PIL
     os.environ['RDKIT_CANVAS'] = 'sping'
     mol = Chem.MolFromSmiles('c1([O])ccc(O)cc1')
     img = Draw.MolToImage(mol)
     self.assertTrue(img)
+    # img.show()
+
+  def testSpecialCases(self):
+    options = Draw.DrawingOptions()
+    options.atomLabelDeuteriumTritium = True
+    self._testMolToImage(mol=Chem.MolFromSmiles('[2H][C@]([3H])(C)F'), options=options)
+    # shared rings
+    self._testMolToImage(mol=Chem.MolFromSmiles('c1cccc2cc(cccc3)c3cc21'))
+    self._testMolToImage(mol=Chem.MolFromSmiles('C1=CC=CC=CC=C1'))
+    self._testMolToImage(mol=Chem.MolFromSmiles('C=C=C'))
+    self._testMolToImage(mol=Chem.MolFromSmiles('CC#N'), showImage=True)
+    self._testMolToImage(mol=Chem.MolFromSmiles('[CH2-][C-2]C[CH3+][CH5+2]'))
+    self._testMolToImage(mol=Chem.MolFromSmiles('[Na+].[OH-]'))
+    self._testMolToImage(mol=Chem.MolFromSmiles('c1ccccc1c1ccccc1'),
+                         highlightAtoms=(0, 1, 2, 3, 4, 5, 6))
+    self._testMolToImage(mol=Chem.MolFromSmiles('c1ccccc1c1ccccc1'),
+                         highlightBonds=(0, 2, 4, 6, 8, 10))
+    self._testMolToImage(mol=Chem.MolFromSmiles('c1ccccc1c1ccc(cc1)c1ccc(cc1)c1ccc(cc1)'))
```

Comment:
These added tests allowed the refactoring of the bond drawing code by visual inspection.